linera-base/src/crypto/secp256k1/evm.rs

```diff
@@ -391,14 +401,33 @@ impl EvmSignature {
     }
 
     /// Checks a signature.
-    pub fn check<'de, T>(&self, value: &T, author: &EvmPublicKey) -> Result<(), CryptoError>
+    pub fn check<'de, T>(&self, value: &T, author: EvmPublicKey) -> Result<(), CryptoError>
     where
         T: BcsSignable<'de> + fmt::Debug,
     {
         let prehash = CryptoHash::new(value).as_bytes().0;
         self.verify_inner::<T>(prehash, author)
     }
 
+    /// Checks a signature against a recovered public key.
+    pub fn check_with_recover<'de, T>(
+        &self,
+        value: &T,
+        sender_address: [u8; 20],
+    ) -> Result<(), CryptoError>
+    where
+        T: BcsSignable<'de> + fmt::Debug,
+    {
+        let prehash = CryptoHash::new(value).as_bytes().0;
+        let public_key = EvmPublicKey(self.0.recover_from_msg(prehash).unwrap());
+        if public_key.address().0 != sender_address {
+            return Err(CryptoError::InvalidSignature {
+                error: "Recovered public key does not match sender address".to_string(),
+                type_name: T::type_name().to_string(),
+            });
+        }
+        self.verify_inner::<T>(prehash, public_key)
+    }
     /// Verifies a batch of signatures.
```

Comment:
```suggestion

    /// Verifies a batch of signatures.
```

Reply:
Done in [Include EVM address in EvmSignature explicitly](https://github.com/linera-io/linera-protocol/pull/4088/commits/e0240d1af37378bdf380aa53c4b5ce204d0ce088)